src/main/java/in/twizmwaz/cardinal/command/TeamCommands.java

```diff
@@ -108,6 +112,64 @@ public static void myTeam(final CommandContext cmd, CommandSender sender) throws
             sender.sendMessage(new UnlocalizedChatMessage(ChatColor.GRAY + "{0}", new LocalizedChatMessage(ChatConstant.GENERIC_ON_TEAM, team.get().getCompleteName())).getMessage(((Player) sender).getLocale()));
         }
     }
+    @Command(aliases = {"register"}, desc = "Registers an OCN team to play a match", min = 1, max = 1, usage = "<ocnteam>")
+    @CommandPermissions("cardinal.register")
+    public static void register(CommandContext cmd, CommandSender sender) throws CommandException {
+        String ocnteam = cmd.getString(0);
+        TeamModule usedTeam = null;
+        ArrayList registeredTeams = GameHandler.getGameHandler().getMatch().getModules().getModule(TeamRegisterModule.class).getRegisteredTeams();
+        Map<String, TeamModule> usedTeams = GameHandler.getGameHandler().getMatch().getModules().getModule(TeamRegisterModule.class).getUsedTeams();
+        if (registeredTeams.contains(ocnteam)) {
+            throw new CommandException(new LocalizedChatMessage(ChatConstant.ERROR_TEAM_ALREADY_REGISTERED).getMessage(ChatUtil.getLocale(sender)));
+        }
+        if((Teams.getTeams().size() - 1 == usedTeams.size())) {
+            throw new CommandException(new LocalizedChatMessage(ChatConstant.ERROR_CANT_REGISTER_MORE_TEAMS).getMessage(ChatUtil.getLocale(sender)));
+        }
+        for (TeamModule team : Teams.getTeams()) {
+            if (!team.isObserver() && !usedTeams.containsKey(team.getName())) {
+                usedTeam = team;
+                try {
+                    Authenticator.downloadTeamDocument(ocnteam);
+                } catch (IOException e) {
+                    throw new CommandException(new LocalizedChatMessage(ChatConstant.ERROR_CANT_GET_TEAM_LIST).getMessage(ChatUtil.getLocale(sender)));
+                }
+                usedTeams.put(ocnteam, team);
+                usedTeam.setName(ocnteam);
+                Bukkit.getPluginManager().callEvent(new TeamNameChangeEvent(usedTeam));
+                registeredTeams.add(ocnteam);
+                sender.sendMessage(new UnlocalizedChatMessage(ChatColor.GREEN + "{0}",  new LocalizedChatMessage(ChatConstant.GENERIC_TEAM_REGISTERED, ChatColor.GOLD + ocnteam)).getMessage(ChatUtil.getLocale(sender)));
+                for(Player p : Bukkit.getOnlinePlayers()) {
+                    GameHandler.getGameHandler().getMatch().getModules().getModule(TeamRegisterModule.class).addRegisteredPlayer(p.getName(), ocnteam);
+                }
+                return;
+            }
+        }
+    }
+    @Command(aliases = {"unregister"}, desc = "Unregisters an OCN team on match end", min = 1, max = 1, usage = "<ocnteam>")
+    @CommandPermissions("cardinal.register")
+    public static void unregister(CommandContext cmd, CommandSender sender) throws CommandException {
+        String team = cmd.getString(0);
```

Comment:
you should support "/team register Random team" and then "/team unregister ran" should unregister it if it's the only team starting with "ran"